sdktests/common_tests_events_contexts.go

```diff
@@ -169,23 +145,26 @@ func (c CommonEventTests) EventContexts(t *ldtest.T) {
 	dataSource := NewSDKDataSource(t, data)
 
 	for _, p := range makeEventContextTestParams() {
+		outputMatcher := func(context ldcontext.Context) m.Matcher {
+			expectedContext := context
+			if p.outputContext != nil {
+				expectedContext = p.outputContext(context)
+			}
+			return JSONMatchesEventContext(expectedContext, p.redactedShouldBe)
+		}
+
 		t.Run(p.name, func(t *ldtest.T) {
 			contexts := p.contextFactory("doServerSideEventContextTests")
 			events := NewSDKEventSink(t)
-			client := NewSDKClient(t, WithEventsConfig(p.eventsConfig), dataSource, events)
+			client := NewSDKClient(t, c.baseSDKConfigurationPlus(WithEventsConfig(p.eventsConfig), dataSource, events)...)
 
 			c.discardIdentifyEventIfClientSide(t, client, events) // client-side SDKs always send an initial identify
 
-			maybeWithIndexEvent := func(matchers ...m.Matcher) []m.Matcher {
-				// Server-side SDKs send an index event for each never-before-seen user. Client-side SDKs do not.
-				if c.isClientSide {
-					return matchers
-				}
-				return append([]m.Matcher{IsIndexEvent()}, matchers...)
-			}
-
 			t.Run("debug event", func(t *ldtest.T) {
 				context := contexts.NextUniqueContext()
+				if c.isClientSide {
+					client.SendIdentifyEvent(t, context)
+				}
```

Comment:
For a client-side SDK we have to set the current user before doing the evaluation.